crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -147,511 +144,225 @@ async fn process_task(
+        cc_id: CrossChainId {
+            chain: execute_task.message.source_chain,
+            id: execute_task.message.message_id.0,
+        },
+        source_address: execute_task.message.source_address,
+        destination_chain: metadata.name_of_the_solana_chain.clone(),
+        destination_address: execute_task.message.destination_address,
+        payload_hash: execute_task
+            .message
+            .payload_hash
+            .try_into()
+            .unwrap_or_default(),
+    };
+    let command_id = command_id(&message.cc_id.chain, &message.cc_id.id);
+    let (gateway_incoming_message_pda, ..) =
+        axelar_solana_gateway::get_incoming_message_pda(&command_id);
+
+    // For compatibility reasons with the rest of the Axelar protocol we need add custom handling
+    // for ITS & Governance programs
+    let destination_address = message.destination_address.parse::<Pubkey>()?;
+    match destination_address {
+        axelar_solana_its::ID => {
+            // todo ITS specific handling
+            tracing::error!("ITS program not yet supported");
+        }
+        axelar_solana_governance::ID => {
+            // todo Governance specific handling
+            tracing::error!("governance program not yet supported");
+        }
+        _ => {
+            // this is a security check, because the relayer is a signer, we don't want to
+            // sign a tx where a malicious destination contract could drain the account. This is
+            // because the `AxelarMessagePayload` defines an interface where the accounts get
+            // dynamically appended, thus it could also include the relayers account.
+            if let Ok(decoded_payload) = AxelarMessagePayload::decode(&payload) {
+                let relayer_signer_acc_included = decoded_payload
+                    .account_meta()
+                    .iter()
+                    .any(|acc| acc.pubkey == signer);
+                if relayer_signer_acc_included {
+                    eyre::bail!(
+                        "relayer will not execute a transaction where its own key is included"
+                    );
                 }
-                .instrument(tracing::info_span!(
-                    "registering command PDA",
-                    ?approved_message_pda
-                ))
-                .in_current_span();
-
-                Some(output)
-            })
-            .collect::<FuturesUnordered<_>>();
+            }
 
-        let mut command_accounts = Vec::new();
-        while let Some(result) = future_set.next().await {
-            let pubkey = result?;
-            command_accounts.push(pubkey);
+            // if security passed, we broadcast the tx
+            let ix = axelar_executable::construct_axelar_executable_ix(
+                message,
+                &payload,
+                gateway_incoming_message_pda,
+            )?;
+            send_tx_parse_error(solana_rpc_client, keypair, ix).await?;
```

Comment:
if the tx cannot be sent (we get an error), this needs to be reported back to the Amplifier API.
https://github.com/eigerco/solana-axelar-internal/issues/504